asm_riscv/src/lib.rs

```diff
@@ -519,32 +519,50 @@ impl From<I> for u32 {
         }
     }
 }
-
-impl From<u32> for I {
+#[derive(Debug, Clone, Copy)]
+///Error types when converting ``u32`` to ``I`
+pub enum ConversionErrorType {
+    ///Unknown funct3 field
+    UnknownFunct3(u32),
+    ///Unknown funct3 or funct7 field
+    UnknownFunct3Funct7(u32, u32),
+    ///Unknown Environment Control Transfer
+    UnknownEnvCtrlTransfer,
+    ///Unknown opcode
+    UnknownOpcode(u32),
+}
+impl TryFrom<u32> for I {
+    type Error = ConversionErrorType;
```

Comment:
This mostly looks good to me, just a couple things I would like changed before merging.
 
 - Markdown surrounds code blocks with single backticks
 - `ConversionErrorType` could just be `ConversionError`
 - Minor formatting changes - not as important, would merge without (`///` always followed by space, empty line between items)

```suggestion

#[derive(Debug, Clone, Copy)]
/// Error types when converting `u32` to `I`
pub enum ConversionError {
    /// Unknown funct3 field
    UnknownFunct3(u32),
    /// Unknown funct3 or funct7 field
    UnknownFunct3Funct7(u32, u32),
    /// Unknown Environment Control Transfer
    UnknownEnvCtrlTransfer,
    /// Unknown opcode
    UnknownOpcode(u32),
}

impl TryFrom<u32> for I {
    type Error = ConversionError;
    
```